### ISSUE TITLE
govc: Unset MAC address / change addressType

### ISSUE THREAD

**Issue by lukasluecke** [vmware/govmomi#2116]
#282 added the `net.address` parameter to `vm.network.change`, which allows setting a MAC address. Is there a way to do the opposite, i.e. remove the MAC address, and go back to `generated` / `assigned` for `addressType`? Maybe allowing these as magic string values and changing the addressType respectively?

**Comment by dougm** [comment]
Haven't tried it, but this might work:
```diff
modified   govc/flags/network.go
@@ -143,5 +143,8 @@ func (flag *NetworkFlag) Change(device types.BaseVirtualDevice, update types.Bas
 
 	if changed.AddressType != "" {
 		current.AddressType = changed.AddressType
+		if current.AddressType == string(types.VirtualEthernetCardMacTypeGenerated) {
+			current.AddressType = ""
+		}
 	}
 }
```

**Comment by semora81** [comment]
Is there an example of how to use this from the command line? i'm trying to do the exact same thing where I need to set a vm's addresstype back to "automatic" 

thanks

**Comment by semora81** [comment]
#2265 worked like a charm, i've got the call setup as : 
```
govc vm.network.change -vm=testvm -net "Management" -net.address="" ethernet-0       
```
it has set the value to automatic and at next startup of the vm, it auto assigned a mac.

**Comment by dougm** [comment]
Fixed by #2265
